### ISSUE TITLE
Raw histogram very different and less insightful than e.g. RawDigger

### ISSUE THREAD

**Issue by Thanatomanic** [Beep6581/RawTherapee#5899]
Related thread: https://discuss.pixls.us/t/puzzling-rawtherapee-raw-histogram/20006

As Alberto mentions, there is some scaling going on which may not be necessary at all. This is issue is here to revisit the (raw) histogram plotting.

**Comment by Thanatomanic** [comment]
The log-transformation of the histogram happens after the binning of the pixels. It should be the other way around for a properly detailed histogram on the left-hand side.

There is a difficulty with implementing that, because click-drag currently changes the compression factor for the histogram (i.e. the actual transformation that needs to be applied). This means that every change would require a complete recalculation of the histogram, which will probably be slow. The best (only?) alternative seems to be to always _calculate_ a log-histogram in base 2, and still _show_ it with a modifiable compression.

**Comment by heckflosse** [comment]
Isn't the real problem that currently the raw histogram has only 256 bins?

**Comment by Thanatomanic** [comment]
@heckflosse I don't think so. Here's a comparison of `amsterdam.pef` with three histograms, all with 256 bins. First linear-linear, then log2-linear (first transformed, then binned), then log2-linear (first binned, then transformed - current RT situation).

![image](https://user-images.githubusercontent.com/6567747/91970175-8b943c80-ed17-11ea-81b0-54098eaf1949.png)

Clearly the second option is better than the third.

**Comment by heckflosse** [comment]
@Thanatomanic I meant the following:

Why do we squeeze the data into 256 bins a priori (in getRAWHistogram()) at all? Wouldn't it be simpler and clearer to use 65536 bins and pass these to the presentation layer, which then could do with this values what it needs to do?

**Comment by Thanatomanic** [comment]
Ah, that would probably work. Should be plenty of bins for floating point images as well.

**Comment by heckflosse** [comment]
@Thanatomanic Here's a float raw in [0;1] range for testing
[D700_20111013_8870-8872.zip](https://github.com/Beep6581/RawTherapee/files/5162091/D700_20111013_8870-8872.zip)

**Comment by Thanatomanic** [comment]
@heckflosse Thanks, but I'm afraid my toolbox doesn't support floating point dng's, so I cannot analyse it 🙃

**Comment by heckflosse** [comment]
@Thanatomanic It was meant for testing your upcoming improvements to RT histogram ;-)

**Comment by heckflosse** [comment]
Having more than 256 bins also would allow to draw _nice_ histograms wider than 256 pixels...
